arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c

```diff
@@ -235,6 +235,322 @@ static inline bool IRAM_ATTR stack_is_psram(void)
 }
 #endif
 
+/**
+ * From: esp-idf/components/hal/esp32s3/include/hal/spi_flash_encrypted_ll.h
+ */
+
+/**
+ * Choose type of chip you want to encrypt manully
+ */
+
+typedef enum
+{
+    FLASH_ENCRYPTION_MANU = 0, /* !< Manually encrypt the flash chip. */
+    PSRAM_ENCRYPTION_MANU = 1  /* !< Manually encrypt the psram chip. */
+} flash_encrypt_ll_type_t;
+
+/**
+ * Enable the flash encryption function under spi boot mode and download boot
+ * mode.
+ */
```

Comment:
This function header is not compliant with NuttX coding standards. Please check the [`Function header sections. `](https://nuttx.apache.org/docs/latest/contributing/coding_style.html#function-headers) section.